### PR TITLE
Chore: Use object-shorthand batch 2 (refs #6407)

### PR DESCRIPTION
**Pull request by kaicataldo**
First batch: https://github.com/eslint/eslint/pull/6893

Breaking this up into batches as per the discussion here: https://github.com/eslint/eslint/pull/6812

This is the second batch (everything in `/lib` except `/lib/rules`). I'll actually enable the rule in the final batch :)

@eslint/eslint-team Will merge this after I get a few pairs of eyes on this. Discussed with @nzakas in Gitter and we decided to get these reviewed and merged without having to wait the full two days. Since this touches so many files, waiting even a few days can lead to merge conflicts.

Ported from the original PR:
Pretty straightforward, aside from two things:
 *  Had to move a `/* istanbul ignore next */` comment to work for the shorthand notation (see [this issue in the Istanbul repo](https://github.com/gotwarlost/istanbul/issues/609) for more info)
 * There's a bug in Node v4.x.x when using `get` as a shorthand property name. We actually saw [a bug report](https://github.com/eslint/eslint/issues/6408) for this very situation and decided against fixing it, so my solution was to just rename the function. Will comment the specific location in the code.

### CODE REVIEW

File: lib/config/config-initializer.js [hunk 1]
```diff
@@ -479,9 +479,9 @@ function promptUser(callback) {
 //------------------------------------------------------------------------------
 
 const init = {
-    getConfigForStyleGuide: getConfigForStyleGuide,
-    processAnswers: processAnswers,
-    initializeConfig: /* istanbul ignore next */ function(callback) {
+    getConfigForStyleGuide,
+    processAnswers,
+    /* istanbul ignore next */initializeConfig(callback) {
```
Comment (kaicataldo) [inline lib/config/config-initializer.js]: Istanbul comment fix

File: lib/rules.js [hunk 2]
```diff
@@ -65,7 +65,7 @@ function importPlugin(pluginRules, pluginName) {
  * @param {string} ruleId Rule id (file name).
  * @returns {Function} Rule handler.
  */
-function get(ruleId) {
+function getHandler(ruleId) {
```
Comment (kaicataldo) [inline lib/rules.js]: Function renamed to avoid Node v4.x.x bug that treats shorthand properties named get as syntax errors.

**Comment by ilyavolodin** [comment]
LGTM. I would hold off on the next batch, since it looks like we will have to do a patch release on Monday, and it's better to do patch releases with minimal amount of changes.

**Comment by kaicataldo** [comment]
@ilyavolodin Will do -thanks!

**Pull request merged**